promql/functions.go

```diff
@@ -784,8 +784,42 @@ func funcMadOverTime(vals []parser.Value, args parser.Expressions, enh *EvalNode
 	}), annos
 }
 
+// === ts_of_last_over_time(Matrix parser.ValueTypeMatrix) (Vector, Notes)  ===
+func funcTsOfLastOverTime(vals []parser.Value, _ parser.Expressions, enh *EvalNodeHelper) (Vector, annotations.Annotations) {
+	el := vals[0].(Matrix)[0]
+
+	var tf float64
+	if len(el.Floats) > 0 {
+		tf = float64(el.Floats[len(el.Floats)-1].T / 1000)
```

Comment:
```suggestion
		tf = float64(el.Floats[len(el.Floats)-1].T) / 1000
```

Otherwise everything will be rounded down to full seconds. 

It would be good to cover this case in the tests.